packages/blockchain-link/src/workers/base.ts

```diff
@@ -35,11 +35,14 @@ const shuffleEndpoints = (a: string[]) => {
     return a;
 };
 
-export class BaseWorker<API> {
+export type BaseAPI = {
+    isConnected: () => boolean | undefined;
```

Comment:
i know it's currently irrelevant, just thinking out loud. What if some future api will not have this method or method will be named differently?
possible solutions:
- (ugly) make this optional `isConnected?: () => boolean | undefined`
- (preferred) create `isConnected` method in BaseWorker and implement/override it in each worker index, same as `tryConnect`. This way BaseWorker doesnt have to force API type and will not work `api` reference at all

Reply:
You're right, preferred way implemented. Moreover, I made `BaseWorker` class abstract, hopefully without any problems.